src/transformers/file_utils.py

```diff
@@ -496,3 +496,50 @@ def _resumable_file_manager():
             json.dump(meta, meta_file)
 
     return cache_path
+
+
+class cached_property(property):
+    """
+    Descriptor that mimics @property but caches output in member variable.
+
+    From tensorflow_datasets
+
+    Built-in in functools from Python 3.8.
+    """
+
+    def __get__(self, obj, objtype=None):
+        # See docs.python.org/3/howto/descriptor.html#properties
+        if obj is None:
+            return self
+        if self.fget is None:
+            raise AttributeError("unreadable attribute")
+        attr = "__cached_" + self.fget.__name__
+        cached = getattr(obj, attr, None)
+        if cached is None:
+            cached = self.fget(obj)
+            setattr(obj, attr, cached)
+        return cached
+
+
+def torch_required(func):
```

Comment:
I would call the other one `skip_if_no_torch` and this one `raise_if_no_torch`/ `require_torch`. This way the names are too similar for a reader to understand the subtlelty.

Reply:
Hmm, I think they won't be exposed outside of the library so not sure if it's a big deal